re_data_ui/src/partials/Graph.tsx

```diff
@@ -0,0 +1,328 @@
+            to: modelId,
+            arrows: 'to',
+          };
+          graph.edges.push(edge);
+        }
+      });
+    }
+    if (modelChildNodes.length) {
+      modelChildNodes.forEach((child) => {
+        const childDetails = allNodes[child];
+        const {
+          database, schema, name,
+          resource_type: resourceType,
+        } = childDetails;
+        if (supportedResTypes.has(resourceType)) {
+          const childModelId = `${database}.${schema}.${name}`.toLowerCase();
+          graph.nodes.push({
+            id: childModelId,
+            label: name,
+            shape: 'box',
+            color: resourceTypeColors[resourceType],
+          });
+
+          const edge: VisEdge = {
+            from: modelId,
+            to: childModelId,
+            arrows: 'to',
+          };
+          graph.edges.push(edge);
+        }
+      });
+    }
+  } else {
+    Object.entries(allNodes).forEach(([, details]) => {
+      if (supportedResTypes.has(details.resource_type) && details.package_name !== 're_data') {
+        const modelId = `${details.database}.${details.schema}.${details.name}`.toLowerCase();
+        const node: VisNode = {
+          id: modelId,
+          label: details.name,
+          shape: 'box',
+          color: {
+            background: resourceTypeColors[details.resource_type],
+          },
+        };
+        graph.nodes.push(node);
+
+        if (details.resource_type !== 'source') {
+          const d = details as DbtNode;
+          const parentNodes = new Set(d.depends_on.nodes);
+          parentNodes.forEach((parent) => {
+            const parentNode: DbtNode | DbtSource = dbtNodes[parent]
+              ? dbtNodes[parent]
+              : dbtSources[parent];
+            if (parentNode) {
+              const parentModelId = `${parentNode.database}.${parentNode.schema}.${parentNode.name}`.toLowerCase();
+              const edge: VisEdge = {
+                from: parentModelId,
+                to: modelId,
+                arrows: 'to',
+              };
+              graph.edges.push(edge);
+            }
+          });
+        }
+      }
+    });
+  }
+
+  return graph;
+};
+
+export interface GraphViewProps {
+  modelName?: string | null;
+  showModelDetails?: boolean
+}
+
+const GraphView: React.FC<GraphViewProps> = (props: GraphViewProps): ReactElement => {
+  const {
+    modelName = null,
+    showModelDetails = true,
+  } = props;
+  const overview: OverviewData = useContext(RedataOverviewContext);
+  const [graph, setGraph] = useState<IGraph>({
+    nodes: [],
+    edges: [],
+  });
+
+  const updateGraph = async () => {
+    if (overview.graph) {
+      if (modelName) {
+        setGraph(generateGraph(overview, modelName));
+      } else if (!modelName) {
+        setGraph(generateGraph(overview));
+      }
+    }
+  };
+
+  useEffect(() => {
+    (async function anyNameFunction() {
+      await updateGraph();
+    }());
+  }, [modelName, overview]);
```

Comment:
what's the reason behind using an `IIFE` here?

Reply:
removed, was testing a different approach...